src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
+
+  return sum / children.length;
```

Comment:
we can use `reduce` array method here. 

try to understand this code

```suggestion
    const sum = children.reduce((acc, ch) => {
    const mother = people.find(m => m.name === ch.mother);
    const ageDiff = ch.born - mother.born;
    return acc + ageDiff;
  }, 0);

  return sum / children.length;

```